### PR TITLE
DM-39857: Remove --flake8 option and add ruff config

### PR DESCRIPTION
**Pull request by timj**
## Checklist

- [ ] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/ctrl/mpexec/cli/opt/options.py [hunk 1]
```diff
@@ -310,17 +310,17 @@
...
-                                            task composition; `uri` to show predicted dataset URIs of
-                                            quanta"""
+                                            ``history=<FIELD>`` to dump configuration history for a field,
+                                            field name is specified as ``[Task::]<PATTERN>;`` ``dump-config``,
+                                            ``dump-config=Task`` to dump complete configuration for a task
+                                            given its label or all tasks; ``pipeline`` to show pipeline
+                                            composition; ``graph`` to show information about quanta;
+                                            ``workflow`` to show information about quanta and their
+                                            dependency; ``tasks`` to show task composition; ``uri`` to show
+                                            predicted dataset URIs of quanta"""
```
Comment (andy-slac) [inline python/lsst/ctrl/mpexec/cli/opt/options.py:323]: Maybe de-indent the whole thing?

Comment (andy-slac) [inline python/lsst/ctrl/mpexec/cli/opt/options.py:323]: Add period?
```suggestion
                                            predicted dataset URIs of quanta."""
```

File: python/lsst/ctrl/mpexec/cli/opt/options.py [hunk 2]
```diff
@@ -310,17 +310,17 @@
...
-                                            `history=<FIELD>` to dump configuration history for a field, field
-                                            name is specified as [Task::]<PATTERN>; `dump-config`,
-                                            `dump-config=Task` to dump complete configuration for a task given
-                                            its label or all tasks; `pipeline` to show pipeline composition;
-                                            `graph` to show information about quanta; `workflow` to show
-                                            information about quanta and their dependency; `tasks` to show
-                                            task composition; `uri` to show predicted dataset URIs of
-                                            quanta"""
+                                            ``history=<FIELD>`` to dump configuration history for a field,
+                                            field name is specified as ``[Task::]<PATTERN>;`` ``dump-config``,
```
Comment (andy-slac) [inline python/lsst/ctrl/mpexec/cli/opt/options.py:317]:
```suggestion
                                            field name is specified as ``[Task::]<PATTERN>``; ``dump-config``,
```

File: tests/test_executors.py [hunk 3]
```diff
@@ -91,11 +91,13 @@ def getReport(self):
         return self.report
 
     def getDataIds(self, field):
-        """Returns values for dataId field for each visited quanta"""
+        """Return values for dataId field for each visited quanta"""
```
Comment (andy-slac) [inline tests/test_executors.py:94]:
```suggestion
        """Return values for dataId field for each visited quanta."""
```

File: tests/test_taskFactory.py [hunk 4]
```diff
@@ -52,16 +54,23 @@ class FakeConnections(PipelineTaskConnections, dimensions=set()):
...
 
 class FakeConfig(PipelineTaskConfig, pipelineConnections=FakeConnections):
+    """Config class used along with fake connections class."""
+
     widget = pexConfig.Field(dtype=float, doc="test", default=1.0)
 
 
 def mockTaskClass():
-    """A class placeholder that records calls to __call__."""
+    """Record calls to ``__call__`.
```
Comment (andy-slac) [inline tests/test_taskFactory.py:63]:
```suggestion
    """Record calls to ``__call__``.
```

**Review by andy-slac — APPROVED** [review]
Looks good, few minor comments.

File: python/lsst/ctrl/mpexec/cli/opt/options.py (hunk 1 above)
Comment (timj) [reply python/lsst/ctrl/mpexec/cli/opt/options.py:323]: Yes, I was trying to work out why it was all like this and failing 😄

Comment (andy-slac) [reply python/lsst/ctrl/mpexec/cli/opt/options.py]: I think that it was initially indented to an opening `(` but then black wrapped the initial line but did not de-indent other lines.

**Pull request merged**